.github/workflows/build_test.yaml

```
@@ -27,13 +27,13 @@ jobs:
     - name: Install Dependencies
       run: |
         python -m pip install --upgrade pip setuptools wheel
-        pip install PyYAML argparse
-        pip install catkin-pkg ros-buildfarm rosdistro nose coverage
-        pip install unidiff
-        pip install rosdep
-        pip install PyGithub
+        python -m pip install -r test/requirements.txt
+    - name: Run Tests
+      run: pytest -s
```

It's probably a good idea to direct collection at a specific subdirectory. I've been bit by broad crawling like this before.

```suggestion
      run: pytest -s test
```